packages/qwik/src/lib/ssgoi.tsx

```diff
@@ -0,0 +1,92 @@
+import { 
+  component$, 
+  Slot, 
+  useStore, 
+  useContextProvider,
+  useVisibleTask$,
+  noSerialize,
+  type NoSerialize
+} from "@builder.io/qwik";
+import type { SsgoiConfig, SsgoiContext as SsgoiContextType } from "./types";
+import { SsgoiContextId } from "./context";
+import { createSggoiTransitionContext } from "@ssgoi/core";
+import * as viewTransitions from "./view-transitions";
+import * as transitions from "./transitions";
+
+// Serializable config structure
+export interface SsgoiSerializableConfig {
+  defaultTransition?: {
+    type: string;
+    options?: Record<string, any>;
+  };
+  transitions?: Array<{
+    from: string;
+    to: string;
+    transition: {
+      type: string;
+      options?: Record<string, any>;
+    };
+    symmetric?: boolean;
+  }>;
+}
+
+interface SsgoiProps {
+  config?: SsgoiSerializableConfig;
+}
+
+// Helper to resolve transition functions from type strings
+const resolveTransition = (transition: { type: string; options?: any }) => {
+  // Check view-transitions first
+  const viewTransition = (viewTransitions as any)[transition.type];
+  if (viewTransition) {
+    return viewTransition(transition.options || {});
+  }
+  
+  // Check regular transitions
+  const regularTransition = (transitions as any)[transition.type];
+  if (regularTransition) {
+    return regularTransition(transition.options || {});
+  }
+  
+  console.warn(`Unknown transition type: ${transition.type}`);
+  return undefined;
+};
+
+export const Ssgoi = component$<SsgoiProps>(({ config }) => {
+  const contextStore = useStore<{ 
+    context?: NoSerialize<SsgoiContextType>,
+    config?: NoSerialize<SsgoiConfig>
+  }>({});
+
+  useVisibleTask$(() => {
+    // Convert serializable config to real config with functions
```

Comment:
do this lazily when calling getTransition instead. That way, no need for the visible task.